docs/internals/optimizer.rst

```diff
@@ -1018,6 +1047,8 @@ a structural level:
 
 This component uses the DataflowAnalyzer.
 
+LiteralRematerialiser should be run before this step.
```

Comment:
it destroys the expression-split form e.g. if the condition of an `if`-statement is a function call but the body is empty.
`if (iszero(a)) {}` -> `pop(iszero(a))`